scripts/symbolize.py

```diff
@@ -0,0 +1,177 @@
+
+    def spawn_addr2line(self):
+        if not self._addr2line:
+	    elf = self.get_elf(self._bin);
+	    if not elf:
+		return
+	    if self._arch == 'arm':
+		cmd = 'arm-linux-gnueabihf-addr2line'
+	    elif self._arch == 'aarch64':
+		cmd = 'aarch64-linux-gnu-addr2line'
+	    else:
+		return
+	    self._addr2line = subprocess.Popen([cmd, '-f', '-p', '-e', elf],
+						stdin = subprocess.PIPE,
+						stdout = subprocess.PIPE)
+
+    def resolve(self, addr):
+        offs = self._load_addr
+        if int(offs, 0) > int(addr, 0):
+            return '???'
+        reladdr = '0x{:x}'.format(int(addr, 0) - int(offs, 0))
+        self.spawn_addr2line()
+        if not self._addr2line:
+            return '???'
+        try:
+            print >> self._addr2line.stdin, reladdr
+            ret = self._addr2line.stdout.readline().rstrip('\n')
+        except IOError:
+            ret = '!!!'
+        return ret
+
+    def reset(self):
+        self._call_stack_found = False
+        self._load_addr = '0'
+        if self._addr2line:
+            self._addr2line.terminate()
+            self._addr2line = None
+        self._arch = 'arm'
+
+    def write(self, line):
+            if self._call_stack_found:
+                match = re.search(STACK_ADDR_RE, line)
+                if match:
+                    addr = match.group('addr')
+                    pre = match.start('addr')
+                    post = match.end('addr')
+                    self._out.write(line[:pre])
+                    self._out.write(addr)
+                    res = self.resolve(addr)
+                    if self._strip_path:
+                        res = re.sub(re.escape(self._strip_path) + '/*', '',
+                              res)
+                    self._out.write(' ' + res)
+                    self._out.write(line[post:])
+                    return
+                else:
+                    self.reset()
+            match = re.search(CALL_STACK_RE, line)
+            if match:
+                self._call_stack_found = True
+            match = re.search(TA_UUID_RE, line)
+            if match:
+                self._bin = match.group('uuid')
+            match = re.search(TA_INFO_RE, line)
+            if match:
+                self._arch = match.group('arch')
+                self._load_addr = match.group('load_addr')
+            match = re.search(X64_REGS_RE, line)
+            if match:
+                # Assume _arch represents the TEE core. If we have a TA dump,
+                # it will be overwritten later
+                self._arch = 'aarch64'
+            self._out.write(line)
+
+    def flush(self):
+        self._out.flush()
+
+def main():
+    args = get_args()
+    if args.dir:
+        # Flatten list in case -d is used several times *and* with multiple
+        # arguments
+        args.dirs = [item for sublist in args.dir for item in sublist]
+    symbolizer = Symbolizer(sys.stdout, args.dirs, args.strip_path)
+
+    for line in sys.stdin:
```

Comment:
What are you supposed to give on `stdin`? I.e, what input is the script waiting for? Core dump? Stack trace?